wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -45,104 +51,178 @@ class WirePlugin : Plugin<Project> {
+      if (extension.protoPaths.isNotEmpty() ||
+          extension.protoTrees.isNotEmpty() ||
+          extension.protoJars.isNotEmpty()) {
+        protoInput.addTrees(project, extension.protoTrees)
+        protoInput.addJars(project, extension.protoJars)
+        protoInput.addPaths(project, extension.protoPaths)
+      }
+
+      val targets = outputs.map { it.toTarget() }
+      val task =
+          project.tasks.register("generate${source.name.capitalize()}Protos",
+              WireTask::class.java) { task: WireTask ->
+            task.group = GROUP
+            task.description = "Generate protobuf implementation for ${source.name}"
+            task.source(sourceInput.configuration)
+
+            if (task.logger.isDebugEnabled) {
+              sourceInput.debug(task.logger)
+              protoInput.debug(task.logger)
+            }
+            task.outputDirectories = outputs.map { output -> File(output.out!!) }
+            task.sourceInput.set(sourceInput.toLocations())
+            task.protoInput.set(protoInput.toLocations())
+            task.roots = extension.roots.toList()
+            task.prunes = extension.prunes.toList()
+            task.moves = extension.moves.toList()
+            task.sinceVersion = extension.sinceVersion
+            task.untilVersion = extension.untilVersion
+            task.onlyVersion = extension.onlyVersion
+            task.rules = extension.rules
+            task.targets = targets
+            task.permitPackageCycles = extension.permitPackageCycles
+          }
+
+      project.tasks.named(PARENT_TASK).configure {
+        it.dependsOn(task)
+      }
+
+      source.registerTaskDependency(task)
+      for (output in outputs) {
+        // TODO(Benoit) Why do I need this?
+        if (output is JavaOutput) {
+          (project.tasks.named("compileJava") as TaskProvider<JavaCompile>).configure {
+            it.source(output.out!!)
+          }
+        }
       }
-      it.outputDirectories = outputs.map { output -> File(output.out!!) }
-      it.source(sourceInput.configuration)
-      it.sourceInput.set(sourceInput.toLocations())
-      it.protoInput.set(protoInput.toLocations())
-      it.roots = extension.roots.toList()
-      it.prunes = extension.prunes.toList()
-      it.moves = extension.moves.toList()
-      it.sinceVersion = extension.sinceVersion
-      it.untilVersion = extension.untilVersion
-      it.onlyVersion = extension.onlyVersion
-      it.rules = extension.rules
-      it.targets = targets
-      it.permitPackageCycles = extension.permitPackageCycles
     }
+  }
 
-    for (output in outputs) {
-      output.applyToProject(project, wireTask, kotlin)
+  private fun Project.addWireRuntimeDependency(
+    hasJavaOutput: Boolean,
+    hasKotlinOutput: Boolean
+  ) {
+    val isMultiplatform = project.plugins.hasPlugin("org.jetbrains.kotlin.multiplatform")
+    // No need to add a runtime dependency for non-JVM targets.
```

Comment:
Is this comment wrong? In Kotlin multiplatform projects, this code would add the runtime dependency to the `commonMain` source set, which is expected. The project may not have JVM targets though, which is also fine.

Reply:
I'll rephrase